app/_components/HintButton.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from "react";
+import { Button } from "./ui/button";
+
+interface poiId {
+    poi_id: number | undefined;
+  }
+
+const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
+
+const HintButton: React.FC<poiId> = ({ poi_id }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [hints, setHints] = useState<string[]>([]);
+
+  const fetchHints = async () => {
+      try {
+        const response = await fetch(`${BASE_URL}/api/hints/${poi_id}`);
+        const data = await response.json();
+        let arrayOfContent = [];
+        for (let i = 0; i < data.length; i++) {
+            arrayOfContent.push(data[i].content);
+        }
+        setHints(arrayOfContent);
+      } catch (error) {
+        console.error("Error fetching hints:", error);
+      }
+    };
+
+  const toggleDisplay = () => {
+    fetchHints();
+    setIsOpen(!isOpen);
+  };
+
+  const closeHints = () => {
+    toggleDisplay();
+  };
```

Comment:
This is purely an optimization approach, but it might be a good idea to implement a Throttle function to prevent users from spamming the API.

Off the top of my head, it might look something like this:
```
function toggleDisplay(delay = 5000) {
  let shouldThrottle = false;
  let timeoutId = null;
  
  return function throttled(...args) {
    if (shouldThrottle) return;
    
    shouldThrottle = true;
    fetchHints();
    setIsOpen((prev) => !prev);
    
    timeoutId = setTimeout(() => {
      shouldThrottle = false;
    }, delay);
  }
}
```

In this case, `...args` is optional, but you normally use this to pass any additional arguments to the closure that you'd like to. Both of your functions here don't have any arguments passed to them, so it's not technically necessary.